Modules/Intent.Modules.AspNetCore.MultiTenancy/Templates/MultiTenancyConfiguration/MultiTenancyConfigurationTemplatePartial.cs

```diff
@@ -45,34 +46,38 @@ public MultiTenancyConfigurationTemplate(IOutputTarget outputTarget, object mode
                                 methodChainStatement.AddMetadata("add-multi-tenant", true);
                                 methodChainStatement.WithoutSemicolon();
 
-                                if (ExecutionContext.Settings.GetMultitenancySettings().Store().IsEfcore())
+                                switch (ExecutionContext.Settings.GetMultitenancySettings().Store().AsEnum())
```

Comment:
I don't _think_ it's an issue in this case, but just be aware that converting from use of the `.Is...()` to the `.AsEnum()` approach can be functionally different in that if `.AsEnum()` encounters an unknown value (for example an option was removed in a module update), then it throws an exception, while the `.Is...()` approach will not.

Reply:
Totally agree, which is why I want the compiler to inform me when an Enum literal has changed so that I can update it in code.

Reply:
While less rare here, the problem I've encountered with stereotype option properties is when an option has been removed, but the saved metadata still contains it (as it was saved in a previous version). Then during template execution as it tries to work out the enum, it throws an exception. The backwards compatible option in such a scenario is to avoid using `.AsEnum()` and the `.Is..()` methods instead.